src/sources/mysql/mysql-schema.lisp

```diff
@@ -131,23 +105,18 @@
      :finally
      (return schema)))
 
-(defun list-all-indexes (schema
-                         &key
-                           only-tables
-                           including
-                           excluding)
+(defmethod fetch-indexes ((schema schema) (mysql copy-mysql)
+                          &key including excluding)
   "Get the list of MySQL index definitions per table."
   (loop
      :for (table-name name index-type non-unique cols)
      :in (mysql-query (format nil
```

Comment:
This `FORMAT` call, `(format nil (sql X) Y including including excluding excluding)`, repeats itself many times inside the code. Maybe it is worth to extract it into a helper function?

Even more, the call is like `(mysql-query (format ...))`, `(mssql-query (format ...))`, ... Maybe it would be worth to create a generic function and specialize it on the subclasses of `copy-db` that you have nearby?

Reply:
That's a good point. I did a quick improvement here by moving the FORMAT call into the SQL function. I think it makes sense from an abstraction perspective and would make more sense if we ever got to change our templating engine from FORMAT to something else.

I'll mark this comment/conversation as *resolved* for now, please feel free to open an issue and then a PR to improve the situation even more, or see if using cl-yesql would make sense for pgloader.